vsphere/resource_vsphere_content_library.go

```diff
@@ -0,0 +1,76 @@
+package vsphere
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-vsphere/vsphere/internal/helper/contentlibrary"
+)
+
+func resourceVSphereContentLibrary() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceVSphereContentLibraryCreate,
+		Delete: resourceVSphereContentLibraryDelete,
+		Read:   resourceVSphereContentLibraryRead,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The name of the content library.",
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "Optional description of the content library.",
+			},
+			"storage_backing": {
+				Type:        schema.TypeSet,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The name of the content library.",
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+		},
+	}
+}
+
+func resourceVSphereContentLibraryRead(d *schema.ResourceData, meta interface{}) error {
+	c := meta.(*VSphereClient).restClient
+	lib, err := contentlibrary.FromName(c, d.Get("name").(string))
+	if err != nil {
+		return err
```

Comment:
If the library is not found because someone deleted it outside terraform we should be handling it gracefully here. 
I think setting an empty ID and returning nil at that point will remove the object from state. Like 
```suggestion
		if viapi.IsManagedObjectNotFoundError(err) {
			d.SetId("")
			return nil
		}
		return fmt.Errorf("error while searching library %s. Error: %s ", d.Get("name").(string), err)
```